controllers/system/system_controller.go

```diff
@@ -1581,6 +1581,30 @@ func (r *SystemReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 
 	// Update scope from configuration
 	logSystem.V(2).Info("before UpdateConfigStatus", "instance", instance)
+	// Filter out certificates with type other than "ssl_ca"
+	var originalLength int
+	if instance.Spec.Certificates != nil {
+		originalLength = len(*instance.Spec.Certificates)
+		filteredCerts := make([]starlingxv1.CertificateInfo, 0, originalLength)
+		for _, cert := range *instance.Spec.Certificates {
+			if cert.Type == "ssl_ca" {
```

Comment:
1 Can we have the string "ssl_ca" replaced with PlatformCACertificate?
2 Would prefer to have this new section as a new method "clean_deprecated_certificates"
3 Expect a new test against the new method

Reply:
Implemented the `clean_deprecated_certificates` func and added  test case to the method